libcanard/canard.h

```diff
@@ -635,6 +649,54 @@ int8_t canardRxUnsubscribe(CanardInstance* const    ins,
+
+/// Generate an acceptance filter configuration to accept a specific service.
+///
+/// Complex applications will likely subscribe to more service IDs than there are
+/// acceptance filters available in the CAN hardware. In this case, the application
+/// should implement filter consolidation. See canardConsolidateAcceptanceFilterConfigs()
+/// as well as the UAVCAN specification for details.
+///
+/// Users may prefer to instead use a catch-all acceptance filter configuration for accepting
+/// all service requests and responses targeted at the specified local node ID.
+/// See canardMakeAcceptanceFilterConfigForServices() for this.
+CanardAcceptanceFilterConfig canardMakeAcceptanceFilterConfigForService(const CanardPortID service_id,
+                                                                        const CanardNodeID local_node_id);
+
+/// Generate an acceptance filter configuration to accept all service
+/// requests or responses targeted to the specified local node ID.
+///
+/// Due to the relatively low frequency of service transfers expected on a network,
+/// and the fact that a service directed at a specific node is not likely to be rejected by that node,
+/// a user may prefer to use this over canardMakeAcceptanceFilterConfigForService()
+/// in order to simplify the API usage and reduce the number of required hardware CAN acceptance filters.
+CanardAcceptanceFilterConfig canardMakeAcceptanceFilterConfigForServices(const CanardNodeID local_node_id);
+
+/// Consolidate two acceptance filter configurations into a single configuration.
+///
+/// Complex applications will likely subscribe to more subject IDs than there are
+/// acceptance filters available in the CAN hardware. In this case, the application
+/// should implement filter consolidation. While this may make it impossible to create
+/// a 'perfect' filter that only accepts desired subject IDs, the application should apply
+/// consolidation in a manner that minimizes the number of undesired messages that pass
+/// through the hardware acceptance filters and require software filtering (implemented by canardRxSubscribe).
+///
+/// While optimal choice of filter consolidation is a function of the number of available hardware filters,
+/// the set of transfers needed by the application, and the expected frequency of occurence
+/// of all possible distinct transfers on the bus, it is possible to generate a quasi-optimal configuration
+/// if information about the frequency of occurence of different transfers is not known.
+/// For details, see the "Automatic hardware acceptance filter configuration" note under the UAVCAN/CAN section
+/// in the Transport Layer chapter of the UAVCAN specification.
+CanardAcceptanceFilterConfig canardConsolidateAcceptanceFilterConfigs(const CanardAcceptanceFilterConfig* a,
+                                                                      const CanardAcceptanceFilterConfig* b);
```

Comment:
```suggestion
CanardAcceptanceFilterConfig canardConsolidateAcceptanceFilterConfigs(const CanardAcceptanceFilterConfig* const a,
                                                                      const CanardAcceptanceFilterConfig* const b);
```

Same for the definition.